MOM6/extra_sources/mom_cap_gtracer_flux.F90

```diff
@@ -0,0 +1,649 @@
+        if (gas_fluxes%bc(n)%param(1) .le. 0.0) then
+          write (error_string, '(1pe10.3)') gas_fluxes%bc(n)%param(1)
+          call mpp_error(FATAL, ' Bad parameter (' // trim(error_string) //&
+              & ') for land_sea_runoff for ' // trim(gas_fluxes%bc(n)%name))
+        endif
+
+        if (gas_fluxes%bc(n)%implementation .eq. 'river') then
+          do j = jsc,jec
+            do i = isc,iec
+              gas_fluxes%bc(n)%field(ind_flux)%values(i,j) = (1 - ice_fraction(i,j)) *&
+                  & gas_fields_atm%bc(n)%field(ind_runoff)%values(i,j) /&
+                  & gas_fluxes%bc(n)%param(1)
+            enddo
+          enddo
+        else
+          call mpp_error(FATAL, ' Unknown implementation (' //&
+              & trim(gas_fluxes%bc(n)%implementation) // ') for ' // trim(gas_fluxes%bc(n)%name))
+        endif
+      else
+        call mpp_error(FATAL, ' Unknown flux_type (' // trim(gas_fluxes%bc(n)%flux_type) //&
+            & ') for ' // trim(gas_fluxes%bc(n)%name))
+      endif
+    endif
+  enddo
+
+  if (allocated(kw)) then
+    deallocate(kw)
+    deallocate(cair)
+  endif
+end subroutine  atmos_ocean_fluxes_calc
+
+!> Calculate \f$k_w\f$
+!!
+!! Taken from Johnson, Ocean Science, 2010. (http://doi.org/10.5194/os-6-913-2010)
+!!
+!! Uses equations defined in Liss[1974],
+!! \f[
+!!  F = K_g(c_g - H C_l) = K_l(c_g/H - C_l)
+!! \f]
+!! where \f$c_g\f$ and \f$C_l\f$ are the bulk gas and liquid concentrations, \f$H\f$
+!! is the Henry's law constant (\f$H = c_{sg}/C_{sl}\f$, where \f$c_{sg}\f$ is the
+!! equilibrium concentration in gas phase (\f$g/cm^3\f$ of air) and \f$C_{sl}\f$ is the
+!! equilibrium concentration of unionised dissolved gas in liquid phase (\f$g/cm^3\f$
+!! of water)),
+!! \f[
+!!    1/K_g = 1/k_g + H/k_l
+!! \f]
+!! and
+!! \f[
+!!    1/K_l = 1/k_l + 1/{Hk_g}
+!! \f]
+!! where \f$k_g\f$ and \f$k_l\f$ are the exchange constants for the gas and liquid
+!! phases, respectively.
+real function calc_kw(tk, p, u10, h, vb, mw, sc_w, ustar, cd_m)
+  real, intent(in) :: tk !< temperature at surface in kelvin
+  real, intent(in) :: p !< pressure at surface in pa
+  real, intent(in) :: u10 !< wind speed at 10m above the surface in m/s
+  real, intent(in) :: h !< Henry's law constant (\f$H=c_sg/C_sl\f$) (unitless)
+  real, intent(in) :: vb !< Molar volume
+  real, intent(in) :: mw !< molecular weight (g/mol)
+  real, intent(in) :: sc_w
+  real, intent(in), optional :: ustar !< Friction velocity (m/s).  If not provided,
+                                      !! ustar = \f$u_{10} \sqrt{C_D}\f$.
+  real, intent(in), optional :: cd_m !< Drag coefficient (\f$C_D\f$).  Used only if
+                                      !! ustar is provided.
+                                      !! If ustar is not provided,
+                                      !! cd_m = \f$6.1 \times 10^{-4} + 0.63 \times 10^{-4} *u_10\f$
+
+  real :: ra,rl,tc
+
+  tc = tk-273.15
```

Comment:
[Magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) are evil ;)

Reply:
Agreed, but I copied this subroutine from FMScoupler and have deliberately left things as unchanged as possible.

Reply:
No problem. It was not clear where the routine came from.

Reply:
I'll add some more comments throughout